codemods/es-set-tostringtag/index.js

```diff
@@ -0,0 +1,61 @@
+import jscodeshift from 'jscodeshift';
+import { removeImport } from '../shared.js';
+
+/**
+ * @typedef {import('../../types.js').Codemod} Codemod
+ * @typedef {import('../../types.js').CodemodOptions} CodemodOptions
+ */
+
+/**
+ * @param {CodemodOptions} [options]
+ * @returns {Codemod}
+ */
+export default function (options) {
+	return {
+		name: 'es-set-tostringtag',
+		transform: ({ file }) => {
+			const j = jscodeshift;
+			const root = j(file.source);
+			let dirtyFlag = false;
+
+			const { identifier } = removeImport('es-set-tostringtag', root, j);
+
+			root
+				.find(j.CallExpression, {
+					callee: {
+						type: 'Identifier',
+						name: identifier,
+					},
+				})
+				.forEach((path) => {
+					const args = path.value.arguments;
+					const newExpression = j.callExpression(
+						j.memberExpression(
+							j.identifier('Object'),
+							j.identifier('defineProperty'),
+						),
+						[
+							args[0],
+							j.memberExpression(
+								j.identifier('Symbol'),
+								j.identifier('toStringTag'),
+							),
+							j.objectExpression([
+								j.property(
+									'init',
+									j.identifier('configurable'),
+									j.booleanLiteral(true),
+								),
+								// @ts-expect-error
+								j.property('init', j.identifier('value'), args[1]),
+							]),
+						],
+					);
+					j(path).replaceWith(newExpression);
+					dirtyFlag = true;
+				});
+
+			return dirtyFlag ? root.toSource(options) : file.source;
+		},
+	};
+}
```

Comment:
`es-set-tostringtag` sure should be replaced, but it's a little more complex than this logic (third argument and existence check). In most cases, it will work fine, but I remember that there were some potentially dangerous cases.

Reply:
I used the example from the `es-set-tostringtag` docs. Do you know where else I can find examples with more complex logic?

Reply:
[See the source code.](https://github.com/es-shims/es-set-tostringtag/blob/main/index.js)

Reply:
I'm not sure what is the correct way to deal with it. What should the output code look like?

Should I add a condition before calling `Object.defineProperty` like this if force isn't specified?
```
  !Object.hasOwn(tagged, Symbol.toStringTag) &&
    Object.defineProperty(tagged, Symbol.toStringTag, {
      configurable: true,
      value: "new tag",
    });
```
What about If I can't determine the value of `force`?

Reply:
> What about If I can't determine the value of `force`?

If the third arg passed to this function - `(force || !Object.hasOwn(...)) && ...`, if not - just `!Object.hasOwn(...) && ...`.

Reply:
I was able to handle the case where `force` is set to `true` but I wasn't able to handle  more complex cases such as 
```js
const force = true
setToStringTag(obj, 'tagged', { force });
```
or
```js
const options = { force: true }
setToStringTag(obj, 'tagged', options);
```

Reply:
It could be worth doing a github search for that library and seeing the usage. It can give an insight into how many people are using it in a certain way. If not many people are using `force` in the first place, we can leave it for now and try to improve on it later. With codemods there will always be edgecases, not everything is very easy to statically analyze, but if we can get 90% of the usecases covered, and log a warning for cases it wasnt able to analyze/transform, I think thats fine.

Reply:
Yeah, a quick search on https://github.com/search?q=require%28%27es-set-tostringtag+language%3AJavaScript&type=code makes it seem like the `force` isn't being used that much, so we can leave it for now

Reply:
Ok, would you like me to change anything else?

Reply:
I think this LGTM for now. Let's improve the codemods more when we do some battletesting with them on real world projects :)